airbyte-integrations/connectors/source-surveymonkey/unit_tests/test_streams.py

```diff
@@ -0,0 +1,414 @@
+    "per_page": 50,
+    "page": 1,
+    "total": 2,
+    "links": {"self": "https://api.surveymonkey.com/v3/surveys?per_page=50&page=1"},
+}
+
+
+def test_survey_ids(requests_mock):
+    requests_mock.get("https://api.surveymonkey.com/v3/surveys", json=response_survey_ids)
+    stream = SurveyIds(**args_mock)
+    records = stream.read_records(sync_mode=SyncMode.full_refresh)
+    assert list(records) == records_survey_ids
+
+
+def test_user_defined_retry(requests_mock):
+    requests_mock.get(
+        "https://api.surveymonkey.com/v3/surveys",
+        [
+            {
+                "status_code": 429,
+                "headers": {"X-Ratelimit-App-Global-Minute-Remaining": "0"},
+                "json": {
+                    "error": {
+                        "id": 1040,
+                        "name": "Rate limit reached",
+                        "docs": "https://developer.surveymonkey.com/api/v3/#error-codes",
+                        "message": "Too many requests were made, try again later.",
+                        "http_status_code": 429,
+                    }
+                },
+            },
+            {"status_code": 200, "headers": {"X-Ratelimit-App-Global-Minute-Remaining": "100"}, "json": response_survey_ids},
+        ],
+    )
+
+    stream = SurveyIds(**args_mock)
+    stream.default_backoff_time = 3
+    records = stream.read_records(sync_mode=SyncMode.full_refresh)
+    assert list(records) == records_survey_ids
+
+
+# def test_cache_usage(requests_mock):
+#     cache_file_name = source_surveymonkey.streams.cache_file.name
+#     requests_mock.get("https://api.surveymonkey.com/v3/surveys", [
+#           {'status_code': 200,
+#           'headers': {'X-Ratelimit-App-Global-Minute-Remaining': '100'},
+#           'json': response_survey_ids}
+#     ])
+#
+#     stream = SurveyIds(**args_mock)
+#     records = stream.read_records(sync_mode=SyncMode.full_refresh)
+#     assert list(records) == records_survey_ids
+#
+#     records = stream.read_records(sync_mode=SyncMode.full_refresh)
+#     assert list(records) == records_survey_ids
+#
+#     assert requests_mock.call_count == 2
+#
+#
+#     with vcr.use_cassette(cache_file_name, record_mode=None, serializer="json", decode_compressed_response=True) as cass:
+#         # assert cass.requests == 0
+#         # assert cass.responses == 0
+#         assert cass.responses_of('https://api.surveymonkey.com/v3/surveys') == 0
+
+
+def test_slices_from_survey_ids(requests_mock):
+    requests_mock.get("https://api.surveymonkey.com/v3/surveys", json=response_survey_ids)
+    stream_slices = Surveys(**args_mock).stream_slices()
+    assert list(stream_slices) == [{"survey_id": "307785415"}, {"survey_id": "307785388"}]
+
+
+def test_slices_from_config(requests_mock):
+    args = {**args_mock, **{"survey_ids": ["307785415"]}}
+    stream_slices = Surveys(**args).stream_slices()
+    assert list(stream_slices) == [{"survey_id": "307785415"}]
+
+
+response_survey_details = {
```

Comment:
Could we shift big structures to `conftest.py` or `helpers.py`. 

The same for other tests.